pkg/kubelet/nodestatus/setters.go

```diff
@@ -57,25 +57,46 @@ const (
 type Setter func(node *v1.Node) error
 
 // NodeAddress returns a Setter that updates address-related information on the node.
-func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
+func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 	validateNodeIPFunc func(net.IP) error, // typically Kubelet.nodeIPValidator
 	hostname string, // typically Kubelet.hostname
 	hostnameOverridden bool, // was the hostname force set?
 	externalCloudProvider bool, // typically Kubelet.externalCloudProvider
 	cloud cloudprovider.Interface, // typically Kubelet.cloud
 	nodeAddressesFunc func() ([]v1.NodeAddress, error), // typically Kubelet.cloudResourceSyncManager.NodeAddresses
 ) Setter {
+	var nodeIP, secondaryNodeIP net.IP
+	if len(nodeIPs) > 0 {
+		nodeIP = nodeIPs[0]
+	}
 	preferIPv4 := nodeIP == nil || nodeIP.To4() != nil
 	isPreferredIPFamily := func(ip net.IP) bool { return (ip.To4() != nil) == preferIPv4 }
 	nodeIPSpecified := nodeIP != nil && !nodeIP.IsUnspecified()
 
+	if len(nodeIPs) > 1 {
+		if externalCloudProvider || cloud != nil {
+			klog.Warningf("Cloud providers do not support multiple --node-ip values; ignoring second value")
```

Comment:
that is not entirely true. That means that cloud hosted clusters will always have host net with single IP? 

can we double check @andrewsykim would know more.

Reply:
I worded this badly. It's not supposed to be saying anything about what the cloud provider can or can't do, it's just saying that _kubelet_ does not allow you to pass multiple `--node-ip` values if you are using a cloud provider, at this time.

(I assume we will eventually allow dual `--node-ip` with clouds as well, but this is more complicated because, among other things, the behavior of `--node-ip` is currently inconsistent between internal cloud providers and external cloud providers. This is one of the things discussed in KEP-1664 that isn't resolved yet.)

(So, the behavior both before and after this PR is that a cloud provider can return dual-stack node addresses, and kubelet will allow you to override _one_ of them, but not both.)